### PR TITLE
First draft of documentation update for v0.15

### PR DESCRIPTION
**Pull request by sigma-andex**

### CODE REVIEW

File: guides/FFI.md
Comment (JordanMartinez) [inline guides/FFI.md:81]: This is only true for values, not necessarily functions (as the example below breaks this rule).

File: guides/Getting-Started.md
Comment (JordanMartinez) [inline guides/Getting-Started.md:344]: I feel like this is still important to say but for ES modules now.

**Review by JordanMartinez — CHANGES_REQUESTED** [review]
Overall looks good.

Should we create a tag in this repo of its state before any ES modules work gets merged and then link to that for everyone on pre-`0.15.x`?

**Comment by sigma-andex** [comment]
> Overall looks good.
> 
> Should we create a tag in this repo of its state before any ES modules work gets merged and then link to that for everyone on pre-`0.15.x`?

Sounds good

File: guides/Getting-Started.md
Comment (sigma-andex) [reply guides/Getting-Started.md:344]: Ok will add it again with ES modules then

Comment (sigma-andex) [reply guides/Getting-Started.md]: I have added it again and updated to es modules. However I think this paragraph needs a bigger rewrite, it doesn't make much sense anymore imho.

**Comment by JordanMartinez** [comment]
Perhaps we shouldn't merge this until we're getting closer to the v0.15.0 release?

Regardless, I've made the `pre-0.15.0-era` tag.

**Comment by JordanMartinez** [comment]
I think we'll do so immediately after the `0.15.0` release is made. See also #392

**Comment by JordanMartinez** [comment]
Hmm... Then again, this is the first draft, meaning there are other changes we might want to make. And I already have the 0.14-era tag made.

**Comment by sigma-andex** [comment]
@JordanMartinez should we merge this now?

**Comment by JordanMartinez** [comment]
@sigma-andex I don't think so. My main reasons against merging this now is because `0.15.0` isn't out yet. If it gets delayed by some other issue, then this repo's documentation will be incorrect, and it's one of the first things to which we direct new users. 

On the other hand, we don't have a branch where we can iterate on the `0.15.0` updates to this repo. So, I think instead a new branch should be created `0.15.0-updates` into which we merge this and #392. Then, we can iterate on that branch and then merge that branch into `master` once `0.15.0` comes out.

**Pull request merged**
